apps/nextjs-website/src/components/organisms/GitBookContent/components/CodeBlock.tsx

```diff
@@ -1,15 +1,19 @@
-import { Typography } from '@mui/material';
 import { CodeBlockProps } from 'gitbook-docs/markdoc/schema/code';
 import { ReactNode } from 'react';
+import CodeBlockPart from '@/components/molecules/CodeBlockPart/CodeBlockPart';
 
-const CodeBlock = ({ children }: CodeBlockProps<ReactNode>) => (
-  <Typography
-    variant='monospaced'
-    component='div'
-    style={{ wordBreak: 'break-word' }}
-  >
-    {children}
-  </Typography>
+const CodeBlock = ({
+  language,
+  lineNumbers,
+  overflow,
+  children,
+}: CodeBlockProps<ReactNode>) => (
+  <CodeBlockPart
+    code={children as string}
```

Comment:
If the children type is always string, then maybe we should change the type in the `CodeBlockProps`

Reply:
Is there a chance it can be of a different type than `string`?
What if we change these lines like this without removing the generic type parameter from `CodeBlockProps` (all the block props have it)?
```
}: CodeBlockProps<string>) => (
  <CodeBlockPart
    code={children}
```

Reply:
Well, I could ask you the same question in a different way 😄. If you think there is a chance the content can be something else than `string`, would you be still confident in casting that argument?

Otherwise, if you think the children of `CodeBlock` will always be of type `string`, then to me it's ok to change the type, but I would change directly within the `CodeBlockProps` type.